### PR TITLE
8313612: Use JUnit in lib-test/jdk tests

### PR DESCRIPTION
**Pull request by xqoasis**
Modified all tests under lib-test/jdk to use JUnit

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8313612](https://bugs.openjdk.org/browse/JDK-8313612): Use JUnit in lib-test/jdk tests (**Sub-task** - P4)

### Reviewers
 * [Christian Stein](https://openjdk.org/census#cstein) (@sormuras - Committer)
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15131/head:pull/15131` \
`$ git checkout pull/15131`

Update a local copy of the PR: \
`$ git checkout pull/15131` \
`$ git pull https://git.openjdk.org/jdk.git pull/15131/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15131`

View PR using the GUI difftool: \
`$ git pr show -t 15131`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15131.diff">https://git.openjdk.org/jdk/pull/15131.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15131#issuecomment-1678531719)

### CODE REVIEW

**Review by sormuras — APPROVED** [review]

File: test/lib-test/jdk/test/lib/format/ArrayDiffTest.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, 2021, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (johnshajiang) [inline test/lib-test/jdk/test/lib/format/ArrayDiffTest.java:2]: Should it just modify the second year, here is `2021`, to `2023`?

Comment (sormuras) [reply test/lib-test/jdk/test/lib/format/ArrayDiffTest.java:2]: Yes, good catch!

File: test/lib-test/jdk/test/lib/hexdump/HexPrinterTest.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2019, 2020, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (sormuras) [inline test/lib-test/jdk/test/lib/hexdump/HexPrinterTest.java:2]:
```suggestion
 * Copyright (c) 2019, 2023, Oracle and/or its affiliates. All rights reserved.
```

File: test/lib-test/jdk/test/lib/hexdump/ObjectStreamPrinterTest.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2019, 2021, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (sormuras) [inline test/lib-test/jdk/test/lib/hexdump/ObjectStreamPrinterTest.java:2]:
```suggestion
 * Copyright (c) 2019, 2023, Oracle and/or its affiliates. All rights reserved.
```

**Review by sormuras — CHANGES_REQUESTED** [review]
Please only keep first and last/latest year of changes.

File: test/lib-test/jdk/test/lib/hexdump/HexPrinterTest.java [hunk 4]
```diff
@@ -56,35 +60,34 @@ static void testMinimalToStringByteArray() {
...
+                Arguments.of("%5d: ", "%02x", 16, " | ", 50, HexPrinter.Formatters.ofPrimitive(short.class, "%d "), "\n")
+        );
+    }
+
+    static Stream<Arguments> builtinParams() {
+        return Stream.of(
+                Arguments.of("minimal", "", "%02x", 16, "", 64, HexPrinter.Formatters.NONE, ""),
+                Arguments.of("canonical", "%08x  ", "%02x ", 16, "|", 31, HexPrinter.Formatters.PRINTABLE, "|" + System.lineSeparator()),
+                Arguments.of("simple", "%04x: ", "%02x ", 16, " // ", 64, HexPrinter.Formatters.ASCII, System.lineSeparator()),
+                Arguments.of("source", "    ", "(byte)%3d, ", 8, " // ", 64, HexPrinter.Formatters.PRINTABLE,  System.lineSeparator())
```
Comment (turbanoff) [inline test/lib-test/jdk/test/lib/hexdump/HexPrinterTest.java:84]:
```suggestion
                Arguments.of("source", "    ", "(byte)%3d, ", 8, " // ", 64, HexPrinter.Formatters.PRINTABLE, System.lineSeparator())
```

**Comment by liach** [comment]
Should all static `@Test` methods be converted to instance methods, as recommended by JUnit? See https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-definitions

> Test Method
any **instance** method that is directly annotated or meta-annotated with `@Test`, ...

File: test/lib-test/jdk/test/lib/format/ArrayDiffTest.java (hunk 1 above)
Comment (sormuras) [reply test/lib-test/jdk/test/lib/format/ArrayDiffTest.java:2]: The line must only two years. Like so:

` * Copyright (c) 2020, 2023, Oracle and/or its affiliates. All rights reserved.`

File: test/lib-test/jdk/test/lib/hexdump/HexPrinterTest.java [hunk 5]
```diff
@@ -204,25 +208,26 @@ static void testASCII() {
...
-                {int.class, "%08x, "},
-                {long.class, "%16x "},
-                {float.class, "%3.4f "},
-                {double.class, "%6.3g "},
-                {boolean.class, "%b "},
-        };
+        assertEquals(expected, actual, "Formatters.ASCII mismatch");
+    }
+
+    // PrimitiveFormatters
```
Comment (liach) [inline test/lib-test/jdk/test/lib/hexdump/HexPrinterTest.java:214]: This comment is now useless.
```suggestion
```

File: test/lib-test/jdk/test/lib/hexdump/ObjectStreamPrinterTest.java [hunk 6]
```diff
@@ -47,54 +49,53 @@
...
-    @DataProvider(name = "SingleObjects")
-    Object[][] sources() {
-        return new Object[][]{
-                {"A Simple", new A(), 1, 1, 0},
-                {"BNoDefaultRO has no call to defaultReadObject", new BNoDefaultRO(), 2, 1, 1},
-                {"BDefaultRO has call to defaultReadObject", new BDefaultRO(), 2, 1, 1},
-                {"CNoDefaultRO extends BNoDefaultRO with no fields", new CNoDefaultRO(), 3, 1, 3},
-                {"CDefaultRO extends BDefaultRO with no fields", new CDefaultRO(), 3, 1, 3},
-        };
+    // SingleObjects
```
Comment (liach) [inline test/lib-test/jdk/test/lib/hexdump/ObjectStreamPrinterTest.java:90]:
```suggestion
```

File: test/lib-test/jdk/test/lib/format/ArrayDiffTest.java (hunk 1 above)
Comment (sormuras) [inline test/lib-test/jdk/test/lib/format/ArrayDiffTest.java:2]:
```suggestion
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates. All rights reserved.
```

**Review by sormuras — APPROVED** [review]
Looks good to me.

Tested with `--test test/lib-test/jdk/test/lib`.

**Review by asotona — APPROVED** [review]

**Pull request closed without merging**